khiops/utils/dataset.py

```
@@ -29,12 +32,268 @@
+                "They must be different."
+            )
+        for table in (parent_table, child_table):
+            if not table in tables_spec.keys():
+                raise ValueError(
+                    f"Relation #{i} ({parent_table}, {child_table}) contains "
+                    f"non-existent table '{table}'. All relation tables must exist "
+                    "in the 'tables' entry."
+                )
+        if parents_and_children.count(relation[:2]) > 1:
+            raise ValueError(
+                f"Relation #{i} ({parent_table}, {child_table}) occurs "
+                f"{parents_and_children.count(relation[:2])} times. "
+                f"Each relation must be unique."
+            )
+
+        # Check hierachical keys
+        _check_hierarchical_keys(
+            i,
+            parent_table,
+            tables_spec[parent_table][1],
+            child_table,
+            tables_spec[child_table][1],
+        )
+
+    # Check there are no cycles
+    _check_no_cycles(relations_spec, main_table_name)
+
+
+def _check_hierarchical_keys(
+    relation_id, parent_table, parent_table_key, child_table, child_table_key
+):
+    """Check that the parent table's key is contained in the child table's key"""
+    table_key_error = False
+    if isinstance(parent_table_key, str) and isinstance(child_table_key, str):
+        table_key_error = child_table_key != parent_table_key
+    elif isinstance(parent_table_key, str) and is_list_like(child_table_key):
+        table_key_error = parent_table_key not in child_table_key
+    elif is_list_like(parent_table_key) and is_list_like(child_table_key):
+        table_key_error = not set(parent_table_key).issubset(set(child_table_key))
+    elif is_list_like(parent_table_key) and isinstance(child_table_key, str):
+        table_key_error = True
+
+    if table_key_error:
+        if isinstance(child_table_key, str):
+            child_table_key_msg = f"[{child_table_key}]"
+        else:
```

Shouldn't we also check that `child_table_key` is a `list-like` of `str`s? Otherwise, if there is e.g. an `int` in there, then this code would fail; should we let it just fail in `TypeError`?

---

I'll add asserts.

---

Done

---

Did'nt added the asserts in the end, see above.

---

IMHO this is just a low-level Python issue: we can't `join` on lists of `ints`...

---

This is already checked in `_check_table_key`

---

Ok, got it.